### PR TITLE
Improve Solver Debugging

### LINKED ISSUE #676: Better Logging (e.g. file not found error)
We are conveniently using `?` to unwrap expectations in places that seem obvious, but could benefit from a bit more information in places such as the following:

https://github.com/gnosis/dex-services/blob/539efcfd0fecd55fd05430e0ce497d4e2b54bbee/driver/src/price_finding/optimization_price_finder.rs#L306-L313

Reason for this particular example stems from a recent alert bug that took a bit longer to investigate because of the missing attempted file name in the log.

```
mainnet-standard-solver  (occurred in both prod and dev)
    Batch 5285955 failed with unretryable error: 
        No such file or directory (os error 2)
```

If we had known what file was missing, it may have directed us to the problem a bit earlier!

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds additional context to the optimization solver errors and adds a log message to record the exact commandline that was used when invoking the solver.

Closes #676 

### Test Plan

CI

### CODE REVIEW

File: driver/src/price_finding/optimization_price_finder.rs [hunk 1]
```diff
@@ -284,14 +289,21 @@ impl Io for DefaultIo {
...
+                .arg(input_file)
+                .args(&["--outputDir", result_folder])
+                .args(&["--solverTimeLimit", &time_limit])
+                .args(solver_type.to_args());
+
+            debug!("Using solver command `{:?}`", command);
+            command
+        };
+
+        let output = command.output()?;
```
Comment (e00E) [inline driver/src/price_finding/optimization_price_finder.rs:306]: Should this get a context too?

**Review by e00E — APPROVED** [review]

**Review by bh2smith — APPROVED** [review]
Cool!

File: driver/src/price_finding/optimization_price_finder.rs (hunk 1 above)
Comment (nlordell) [reply driver/src/price_finding/optimization_price_finder.rs:306]: The `IO::run_solver` has error context already, and if the exit status is non-zero there is an `error!` log message.

**Comment by josojo** [comment]
Both, the solver type and the command are printed during the initialization of the "main" and the initialization of  `priceFinder`

But I agree that it might be more useful to have it printed every time, so that we don't have to search the beginning.

Maybe you wanna erase the printing at the beginning during the price finder initialization.

**Comment by nlordell** [comment]
> Both, the solver type and the command are printed during the initialization of the "main" and the initialization of priceFinder

The `debug` statement prints the exact command line invocation including parameters like the input file name, output directory, serialized time limit. These change per run of the solver.

**Review by fleupold — APPROVED** [review]

**Pull request merged**